### ISSUE TITLE
RouteProgress.durationRemaining & RouteStepProgress.fractionTraveled return NaN

### ISSUE THREAD

**Issue by m73dev** [mapbox/mapbox-navigation-ios#686]
In specific case (large GPS jump to destination / waypoint), this can crash app.

I managed to fix it by adding a check to see if it was Nan before returning the value.

RouteProgress.durationRemaining:
```
    /**
     Total seconds remaining on all legs.
     */
    public var durationRemaining: TimeInterval {
        let v = route.legs.suffix(from: legIndex + 1).map { $0.expectedTravelTime }.reduce(0, +) + currentLegProgress.durationRemaining
        if v == Double.infinity || v.isNaN {
            return 0
        }else{
            return v
        }
    }
```

RouteStepProgress.fractionTraveled:
```
    /**
     Number between 0 and 1 representing fraction of current step traveled.
     */
    public var fractionTraveled: Double {
        let v = distanceTraveled / step.distance
        
        if v.isNaN || v == Double.infinity {
            return 0
        }else{
            return v
        }
    }
```

**Comment by bsudekum** [comment]
Could you put this into a PR?

**Comment by bsudekum** [comment]
This should be fixed in v0.10.0.
